src/tmpfiles/tmpfiles.c

```diff
@@ -2632,6 +2742,64 @@ static int find_gid(const char *group, gid_t *ret_gid, Hashmap **cache) {
         return name_to_gid_offline(arg_root, group, ret_gid, cache);
 }
 
+static int parse_age_by_from_arg(const char *age_by_str, Item *item) {
+        const char *s = NULL;
+        AgeBy ab_f = 0, ab_d = 0;
+
+        static const struct {
+                char age_by_chr;
+                AgeBy age_by_flag;
+        } age_by_types[] = {
+                { 'a', AGE_BY_ATIME },
+                { 'b', AGE_BY_BTIME },
+                { 'c', AGE_BY_CTIME },
+                { 'm', AGE_BY_MTIME },
+        };
+
+        assert(age_by_str);
+        assert(item);
+
+        s = age_by_str;
+
+        if (isempty(s))
+                return -EINVAL;
+
+        do {
+                /* Ignore whitespace. */
+                if (IN_SET(*s, ' ', '\t'))
+                        continue;
+
+                size_t i;
+                for (i = 0; i < ELEMENTSOF(age_by_types); i++) {
+                        /*
+                         * Check lower-case for files,
+                         * upper-case for directories.
+                         */
+                        if (*s == age_by_types[i].age_by_chr) {
+                                SET_FLAG(ab_f, age_by_types[i].age_by_flag, true);
+                                break;
+                        } else if (*s == ascii_toupper(age_by_types[i].age_by_chr)) {
+                                SET_FLAG(ab_d, age_by_types[i].age_by_flag, true);
+                                break;
+                        }
+                }
+
+                /* Invalid character. */
+                if (i >= ELEMENTSOF(age_by_types))
+                        return -EINVAL;
+        } while (*(++s));
+
+        /* No match. */
+        if (ab_f == 0 && ab_d == 0)
+                return -EINVAL;
+
+        /* Note: "ctime" is ignored for directories. */
+        item->age_by_file = ab_f > 0 ? ab_f : AGE_BY_DEFAULT_FILE;
+        item->age_by_dir = (ab_d > 0 && ab_d != AGE_BY_CTIME) ? ab_d : AGE_BY_DEFAULT_DIR;
```

Comment:
i don't grok this?

Reply:
The reason I went with this is because if the used specifies `C:1m` (`ctime` is the only option specified). Since `ctime` is ignored for directories, I thought using the default value would be appropriate.

Reply:
i think our default should be to ignore ctimes, but if the suer explicitly configures it to be checked we should do as the user says, even though it's stupid. this tool is low-level enough to just do what the user says in case the user went the extra mile to tell is so